app/lib/actions/remote_execution/run_host_job.rb

```diff
@@ -47,8 +47,13 @@ def plan(job_invocation, host, template_invocation, proxy_selector = ::RemoteExe
         action_options = provider.proxy_command_options(template_invocation, host)
                                  .merge(additional_options)
 
-        plan_delegated_action(proxy, provider.proxy_action_class, action_options)
-        plan_self
+        proxy_action = plan_delegated_action(proxy, provider.proxy_action_class, action_options)
+        plan_self(proxy_action_output: proxy_action)
+      end
+
+      def run
+        self.input[:proxy_action_output]
+        self.output.update(success: true)
```

Comment:
Just because we are in a run phase, it doesn't mean the previous step was a success. Maybe using `exit_status.to_s == '0'` instead of hardcoded `true` would be better